src/core/Akka.Persistence/Journal/AsyncWriteJournal.cs

```diff
@@ -370,44 +363,62 @@ void Resequence(Func<IPersistentRepresentation, Exception, object> mapper, IImmu
 
                         foreach (var p in (IEnumerable<IPersistentRepresentation>)aw.Payload)
                         {
-                            _resequencer.Tell(new Desequenced(mapper(p, exception), counter + i + 1, message.PersistentActor, p.Sender));
+                            _resequencer.Tell(new Desequenced(mapper(p, exception), counter + i + 1, message.PersistentActor, p.Sender), self);
                             i++;
                         }
                     }
                     else
                     {
                         var loopMsg = new LoopMessageSuccess(resequencable.Payload, message.ActorInstanceId);
-                        _resequencer.Tell(new Desequenced(loopMsg, counter + i + 1, message.PersistentActor, resequencable.Sender));
+                        _resequencer.Tell(new Desequenced(loopMsg, counter + i + 1, message.PersistentActor, resequencable.Sender), self);
```

Comment:
Since we're now inside an ambient `await`-ed context, we need to pass the closed over version of `Context.Self` as the explicit sender here since that can no longer be done for us implicitly.